.github/workflows/validate-changelog.yml

```
@@ -0,0 +1,31 @@
+name: validate-changelog
+on: pull_request
+
+jobs:
+  validate-changelog:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: poetry install
```

```suggestion
      - name: Checkout
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
         
     - name: Python ${{ matrix.python-version }} - Setup Environment
        uses: ./.github/actions/setup_environment
        with:
          python-version: '3.10'
```

You can call here setup-env and get rid of all the rest to avoid code duplication